metadata-ingestion/src/datahub/ingestion/source/glue.py

```diff
@@ -0,0 +1,211 @@
+        "long": NumberTypeClass,
+        "map": MapTypeClass,
+        "null": NullTypeClass,
+        "set": ArrayTypeClass,
+        "smallint": NumberTypeClass,
+        "string": StringTypeClass,
+        "struct": MapTypeClass,
+        "timestamp": TimeTypeClass,
+        "tinyint": NumberTypeClass,
+        "union": UnionTypeClass,
+        "varchar": StringTypeClass,
+    }
+
+    if field_type in _field_type_mapping.keys():
+        type_class = _field_type_mapping[field_type]
+    elif field_type.startswith("array"):
+        type_class = ArrayTypeClass
+    elif field_type.startswith("map") or field_type.startswith("struct"):
+        type_class = MapTypeClass
+    elif field_type.startswith("set"):
+        type_class = ArrayTypeClass
+    # TODO better handling of unknown type
+    else:
+        type_class = NullTypeClass
+    data_type = SchemaFieldDataType(type=type_class())
+    return data_type
+
+
+# TODO: set this @dataclass
+class GlueSourceConfig(ConfigModel):
+    env: str = "PROD"
+    aws_access_key_id: Optional[str] = None
+    aws_secret_access_key: Optional[str] = None
+    aws_session_token: Optional[str] = None
+    aws_region: Optional[str] = None
+
+    @property
+    def glue(self):
+        if self.aws_access_key_id and self.aws_secret_access_key and self.aws_session_token:
+            return boto3.client(
+                'glue',
+                aws_access_key_id=self.aws_access_key_id,
+                aws_secret_access_key=self.aws_secret_access_key,
+                aws_session_token=self.aws_session_token,
+                region_name=self.aws_region
+            )
+        elif self.aws_access_key_id and self.aws_secret_access_key:
+            return boto3.client(
+                'glue',
+                aws_access_key_id=self.aws_access_key_id,
+                aws_secret_access_key=self.aws_secret_access_key,
+                region_name=self.aws_region
+            )
+        else:
+            return boto3.client(
+                'glue',
+                region_name=self.aws_region
+            )
+
+
+@dataclass
+class GlueSourceReport(SourceReport):
+    tables_scanned = 0
+
+    def report_table_scanned(self) -> None:
+        self.tables_scanned += 1
+
+
+@dataclass
+class GlueSource(Source):
+    source_config: GlueSourceConfig
+    report = GlueSourceReport()
+
+    def __init__(self, config: GlueSourceConfig, ctx: PipelineContext):
+        super().__init__(ctx)
+        self.source_config = config
+        self.report = GlueSourceReport()
+        self.glue = config.glue
+        self.env = config.env
+
+    @classmethod
+    def create(cls, config_dict, ctx):
+        config = GlueSourceConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        logger.debug("listing tables")
+        tables = []
+        kwargs = {}
+        while True:
+            data = self.glue.search_tables(**kwargs)
```

Comment:
I think this scans all the databases and tables? As mentioned above, would be nice if you could skip some based on some params, because this might be too heavy or not really desired behaviour. Then maybe add them to a skipped list in the report or so.

Reply:
yeah we thought about adding it into the config but ran out of time sadly, will look into it again!

Reply:
So maybe here take over this part?
```
       for database in databases:
            if not self.config.schema_pattern.allowed(database["Name"]):
                self.report.report_dropped(database["Name"])
                continue
            tables = get_glue_tables(
                db_name=database["Name"],
                catalog_id=self.config.catalog_id,
                pagination_config={"PageSize": 1000},
            )
            for table in tables:
                if not self.config.table_pattern.allowed(table["Name"]):
                    self.report.report_dropped(table["Name"])
                    break
```

Or try to do a filter with ```data = self.glue.search_tables(**kwargs)```
But I'm not sure if you can avoid making unnecessary calls for the databases you don't want to include.